### PR TITLE
Simplify guides

### PR DESCRIPTION
**Pull request by tyrannokapi**
This pull request:
* Significantly simplifies the process of getting started with Sponge development by consolidating workspace setup information into a shared folder
* Removes redundancies across several documents
* Adds detailed instructions for setting up Maven and Gradle on new plugin projects - for IntelliJ IDEA, Eclipse, and NetBeans
* Fixes several grammar issues and clarifies wording across several documents

Building upon the idea in #58, I moved all workspace setup information into one central folder. For this to make sense, I created "tracks" for four series of guides to help developers get started with the Sponge project. These tracks begin in ``/workspace/getting-started/``.

At the bottom of each guide in the workspace directory, I added links under a heading titled "Track Continuation." This builds upon @BlackGoat's idea of including links to help with the flow of information. 

This PR addresses #58 and #69. Feedback is welcome, particularly on the concept of tracks. I think it's relatively future-proof, but as with anything, there are faults.

### CODE REVIEW

**Comment by gratimax** [comment]
Wow. :+1:

File: source/plugins/quick-start.rst [hunk 1]
```diff
@@ -2,89 +2,81 @@
...
-    }
 
-When loading your plugin .jar file, Sponge will find your plugin class automatically and load it.
+    @Plugin(id = "example", name = "Example Project", version = "1.0")
+    public class ExampleProject {
+        // Woot! This is a comment!
+    }
 
-Be sure to also change the ID and name of your plugin, as well as its version. The ID should be a simple identifier without special characters or spaces, as it is used for generating configuration files and letting users refer to your plugin by its ID.
+Remember to changing the ID, name, and version of your plugin in the annotation. The ID should be a simple identifier with no special characters or spaces, as it is used for generating configuration files and for identification purposes.
```
Comment (Zidane) [inline source/plugins/quick-start.rst]: to *change

**Comment by Zidane** [comment]
Bravo, seriously thank you for the PR.

**Comment by sibomots** [comment]
Indeed.

"Author! Author!"



Sent from my Windows Phone
________________________________

**Comment by tyrannokapi** [comment]
@Zidane Thanks, I'll fix that in a couple minutes.
@boformer Ah, the Gradle Plugin does seem to work after all. It throws an error when it's installing, which is what misled me before. It appears to work with a little bit of time, though.

I'll make a new commit in a couple minutes with the necessary changes.

**Comment by boformer** [comment]
👍

**Comment by simon816** [comment]
You mention installing the git client on Linux and OSX but suggest using the GitHub for Windows application on Windows (personally I don't like it).
It should point users to the standard git client ([msysgit](https://msysgit.github.io/))
There should be a link to the official download page too, http://git-scm.com/download is a good landing page to link.

**Comment by progwml6** [comment]
@simon816 github for windows is not nearly as usable as atlassian's sourcetree or several other clients

**Comment by ryantheleach** [comment]
and sourcetree misses stuff. I'd stick to recommending msys since all the tutorials elsewhere seem to be based around git command line, and working out what a given command is in the gui isn't always straight forward.

**Pull request closed without merging**